Packages/vcs/Lib/VTKAnimate.py

```diff
@@ -0,0 +1,89 @@
+# VTK Backend Animation Module
+## Author:  Charles Doutriaux
+import animate_helper
+import vcs
+import time
+import random
+import hashlib
+import os
+import glob
+
+class VTKAnimationCreate(animate_helper.StoppableThread):
+  def __init__(self, controller):
+    animate_helper.StoppableThread.__init__(self)
+    self.controller = controller
+
+  def run(self):
+    #self.describe()
+    self.controller.animation_created = True
+    self.controller._unique_prefix=hashlib.sha1(time.asctime()+str(random.randint(0,10000))).hexdigest()
+    #print "NFRAMES:",self.controller.number_of_frames()
+
+  def describe(self):
+    for info in self.controller.animate_info:
+      disp = info[0]
+      print "BACKEND:",disp.backend
+      print "TYPE:",disp.g_type
+      print "Name:",disp.g_name
+      if info[1][0] is not None:
+        print "Array:",info[1][0].shape
+      else:
+        print "No Array"
+
+class VTKAnimationPlayback(animate_helper.AnimationPlayback):
+  def __init__(self, controller):
+    animate_helper.AnimationPlayback.__init__(self,controller)
+  pass
+
+class VTKAnimate(animate_helper.AnimationController):
+    def __init__(self,vcs_self):
+        animate_helper.AnimationController.__init__(self,vcs_self)
+        self.AnimationCreate = VTKAnimationCreate
+        self.AnimationPlayback = VTKAnimationPlayback
+        self.cleared = False
+        import atexit
+        atexit.register(self.close)
+    def draw_frame(self):
+      #print "Drawing frame:",self.frame_num,self._unique_prefix
+      png_name=os.path.join(os.environ["HOME"],".uvcdat",self._unique_prefix,"anim_%i.png" % self.frame_num)
+      if os.path.exists(png_name) and len(self.animation_files)==self.number_of_frames():# and self.playback_params.zoom_factor!=1:
+        ## Ok we have the pngs and we need to zoom, need to use png
+        ## maybe the zoom factor thing can be taken off, not sure what's faster
+        if not self.cleared:
+            self.vcs_self.backend.clear()
+            self.cleared = True
+        self.vcs_self.put_png_on_canvas(
+          png_name,
+          self.playback_params.zoom_factor,
+          self.playback_params.vertical_factor,
+          self.playback_params.horizontal_factor)
+      else: # Ok no pngs let's update the arrays and redraw
+        ## Ok let's loop through the arrays and figure out the slice needed and update
+        for i,info in enumerate(self.vcs_self.animate_info):
+          disp,slabs = info
+          slab = slabs[0]
+          if slab is None:
+            continue # nothing to do
+          #Ok we have a slab, let's figure which slice it is
+          args=[]
+          N=1
```

Comment:
Please don't use n and N in the same scope. This has lead to errors in other parts of the code in the past, and is a very bad practice. In general, one letter identifiers and identifiers that differ only in case aren't good ideas -- it can be hard to figure out what short identifiers are, and mixing cases is very error-prone.